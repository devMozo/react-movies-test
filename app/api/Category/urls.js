import { MOVIES_API_KEY } from '../../constants/config';
import { QUERY_API_KEY } from '../../constants/queryStrings';
import { URL_MOVIE_API } from '../config';

export const URL_GET_CATEGORIES = `${URL_MOVIE_API}/genre/movie/list?${QUERY_API_KEY}=${MOVIES_API_KEY}`;
