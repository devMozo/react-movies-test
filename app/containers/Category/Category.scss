.Category {
    * {
        font-family: 'Arial';
    }

    .Category__list {
        padding: 0 20px;
    }
}
