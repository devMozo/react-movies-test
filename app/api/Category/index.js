import { URL_GET_CATEGORIES } from './urls';
import Axios from 'axios';

const getAllCategories = () => {
    return Axios.get(URL_GET_CATEGORIES, {
        headers: {
            'Access-Control-Allow-Origin': '*',
        },
    })
        .then(response => response.data.genres)
        .catch(err => err);
};

export { getAllCategories };
