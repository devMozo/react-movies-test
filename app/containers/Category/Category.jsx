import React from 'react';
import { connect } from 'react-redux';
import HandlerLoaders from '../../components/HandlerLoaders/HandlerLoaders';
import MovieItem from '../../components/MovieItem/MovieItem';
import { getMovies } from '../../redux-app/actions/MovieByCategoryAction';
import './Category.scss';

export class Category extends React.PureComponent {
    getMovies = () => {
        const { getMovies, match } = this.props;
        getMovies(match.params.id, 1);
    };

    componentDidMount() {
        this.getMovies();
    }

    render() {
        const { movies, loading, error } = this.props;

        return (
            <section className="Category">
                <HandlerLoaders loading={loading} error={error} onTryAgain={this.getMovies}>
                    <ul className="Category__list">
                        {movies &&
                            movies.map(movie => (
                                <MovieItem
                                    key={movie.id}
                                    id={movie.id}
                                    title={movie.title}
                                    posterPath={movie.poster_path}
                                    overview={movie.overview}
                                />
                            ))}
                    </ul>
                </HandlerLoaders>
            </section>
        );
    }
}

const mapStateToProps = state => {
    return {
        ...state.movieByCategory,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        getMovies: (genreId, page) => {
            dispatch(getMovies(genreId, page));
        },
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Category);
