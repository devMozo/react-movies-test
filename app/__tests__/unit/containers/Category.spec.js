import React from 'react';
import { shallow, mount } from 'enzyme';
import { Category } from '../../../containers/Category/Category';
import { BrowserRouter as Router } from 'react-router-dom';

const MOCK_DISCOVER_MOVIE = require('../../../__mocks__/discoverMovie.json');

describe('<Category />', () => {
    let PROPS;
    const mockedGetMovies = jest.fn();

    beforeEach(() => {
        PROPS = {
            movies: [],
            loading: false,
            error: false,
            getMovies: mockedGetMovies,
            match: {
                params: {
                    id: 28,
                },
            },
        };
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Category {...PROPS} />);
        expect(component).toMatchSnapshot();
    });

    it('should render all categories when the data was fetched', () => {
        PROPS = {
            ...PROPS,
            movies: MOCK_DISCOVER_MOVIE.results,
        };

        const component = mount(
            <Router>
                <Category {...PROPS} />
            </Router>,
        );

        expect(component.find('.MovieItem').length).toBe(MOCK_DISCOVER_MOVIE.results.length);
        expect(mockedGetMovies).toHaveBeenCalled();
    });
});
