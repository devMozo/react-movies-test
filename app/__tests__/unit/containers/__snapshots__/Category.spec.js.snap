// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> should render correctly in "debug" mode 1`] = `
<section
  className="Category"
>
  <Memo(HandlerLoaders)
    error={false}
    loading={false}
    onTryAgain={[Function]}
  >
    <ul
      className="Category__list"
    />
  </Memo(HandlerLoaders)>
</section>
`;
